File: form-scanner-service/src/main/java/fr/vandriessche/rallyeschema/formscannerservice/entities/StageParam.java
package fr.vandriessche.rallyeschema.formscannerservice.entities;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Document
public class StageParam {
	public StageParam(Integer stage) {
		this.stage = stage;
	}

	@Id
	private String id;

	@Indexed(unique = true)
	private Integer stage;

	private String name;
	private Boolean inactive;

	@DBRef
	private List<ResponseFileParam> responseFileParams = new ArrayList<>();

	private LinkedHashMap<String, QuestionPointParam> questionPointParams = new LinkedHashMap<>();
	private LinkedHashMap<String, PerformancePointParam> performancePointParams = new LinkedHashMap<>();

	private LinkedHashMap<String, QuestionParam> questionParams = new LinkedHashMap<>();

}

File: rallye-schema-response-webapp/src/app/param/models/stage-param.ts
import { HalLinks } from '../../models/hal-links';
import { PerformancePointParam } from './performance-point-param';
import { QuestionParam } from './question-param';
import { QuestionPointParam } from './question-point-param';
import { StageGroup } from './stage-group';

export interface QuestionPointParams {
  [key: string]: QuestionPointParam;
}

export interface PerformancePointParams {
  [key: string]: PerformancePointParam;
}

export interface QuestionParams {
  [key: string]: QuestionParam;
}

export interface StageParam {
  id?: string;
  version?: number;
  stage: number;
  name: string;
  questionPointParams: QuestionPointParams;
  performancePointParams: PerformancePointParams;
  questionParams: QuestionParams;
  group?: StageGroup | null;
  _links?: HalLinks;
}
